tests/build-init-files.py

```diff
@@ -45,7 +49,7 @@ def command(verbose, root_str):
         path
         for tree_root in tree_roots
         for path in root.joinpath(tree_root).rglob("**/")
-        if all(part not in ignores for part in path.parts)
+        if not ignored_path(path) and not all(map(ignored_path, path.iterdir()))
```

Comment:
So we want to include it if it itself is not an ignored path and also it contains a path which isn't ignored.  `any(not ignored_path(item) for item in path.iterdir())`?  I have a strong preference for comprehensions over `map()` and `filter()`.  But this won't handle two layers of dirs with nothing else in either of them.  I think this in part follows from the fact that pycharm goes and uses the file's directory as cwd when running it, combined with us and the tools we use dumping files into cwd.  Otherwise the directories would go away when we switch branches, I think.  Do we want to take one more stop gap step or consider a new design that is much more aware?  Or maybe look at git instead of just directories?  I'm fine either way.